aspnetcore/fundamentals/host/platform-specific-configuration.md

```diff
@@ -0,0 +1,289 @@
+  - `ASPNETCORE_PREVENTHOSTINGSTARTUP` environment variable.
+* To prevent specific hosting startup assemblies from loading, set one of the following to a semicolon-delimited string of hosting startup assemblies to exclude from startup:
+  - [Hosting Startup Exclude Assemblies](xref:fundamentals/host/web-host#hosting-startup-exclude-assemblies) host configuration setting.
+  - `ASPNETCORE_HOSTINGSTARTUPEXCLUDEASSEMBLIES` environment variable.
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.0"
+
+To disable automatic loading of hosting startup assemblies, set one of the following to `true` or `1`:
+
+* [Prevent Hosting Startup](xref:fundamentals/host/web-host#prevent-hosting-startup) host configuration setting.
+* `ASPNETCORE_PREVENTHOSTINGSTARTUP` environment variable.
+
+::: moniker-end
+
+If both the host configuration setting and the environment variable are set, the host setting controls the behavior.
+
+Disabling hosting startup assemblies using the host setting or environment variable disables the assembly globally and may disable several characteristics of an app.
+
+## HostingStartup attribute
+
+A [HostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.hostingstartupattribute) attribute indicates the presence of a hosting startup assembly to activate at runtime.
+
+The entry assembly or the assembly containing the `Startup` class is automatically scanned for the `HostingStartup` attribute. The list of assemblies to search for `HostingStartup` attributes is loaded at runtime from configuration from the [WebHostDefaults.HostingStartupAssembliesKey](/dotnet/api/microsoft.aspnetcore.hosting.webhostdefaults.hostingstartupassemblieskey). The list of assemblies to exclude from discovery is loaded from the [WebHostDefaults.HostingStartupExcludeAssembliesKey](/dotnet/api/microsoft.aspnetcore.hosting.webhostdefaults.hostingstartupexcludeassemblieskey). For more information, see [Web Host: Hosting Startup Assemblies](xref:fundamentals/host/web-host#hosting-startup-assemblies) and [Web Host: Hosting Startup Exclude Assemblies](xref:fundamentals/host/web-host#hosting-startup-exclude-assemblies).
+
+In the following example, the namespace of the hosting startup assembly is `StartupEnhancement`, and the class containing the hosting startup code is `StartupEnhancementHostingStartup`:
+
+[!code-csharp[](platform-specific-configuration/samples-snapshot/2.x/StartupEnhancement.cs?name=snippet1)]
+
+The `HostingStartup` attribute is typically located in the hosting startup assembly's `IHostingStartup` implementation class file.
+
+## Activation approaches
+
+There are three approaches that you can take to set up and activate a hosting startup assembly:
+
+* [NuGet package](#nuget-package-activation)
+* [Class library](#class-library-activation)
+* [Assembly](#assembly-activation)
+
+Each approach is described in the following sections.
+
+## NuGet package activation
```

Comment:
> are not treated specially for this feature

I understand that from the engineering perspective (i.e., delivery is a separate issue), but we're dealing with a slightly complicated scenario overall with some subtle differences between the gross implementation approaches (*gross* = including delivery).

It's easier to organize the topic content for clarity on the differences by going with separate (but similar) sections due the differences in procedure. This leaves the reader simply to ask, "How do I want to supply this?" Then, they read the applicable section. Otherwise if they aren't sure, they can read them in a more SxS fashion and easily compare and contrast the two.

If we roll NuGet into the class lib section (and remove the parts of the sample that pertain to the NuGet approach ... probably water down the text to basic 'use NuGet to deliver it' statements), I think the gross implementation will become *harder to understand.*

It's your call ... I'll cut it if you want, but I don't recommend it. If you want to proceed with cutting/moving, do you want it out of the sample, too?

Just remember as u make your final call, I didn't write this for you ... I wrote it for *ME*. :smile: lol

Reply:
If you keep it then there are some parts that need to be fixed up. E.g. you say to set the environment variable to packageid, but the env var name always needs to use the assembly name which may not match the packageid.